src/functions/util.test.js

```diff
@@ -0,0 +1,60 @@
+const { generateLoginInfo, generateValidLoginInfo } = require("./util")
```

Comment:
It's a good practice to split up the tests in to separate files based on their purpose.

Each test for a particular component needs to be in it's own file.
Integration tests need to be septette from unit tests.
end to end tests need to be separate from both.